discord/team.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-2019 Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from .user import User
+from .asset import Asset
+from .enums import TeamMembershipState, try_enum
+
+
+class Team:
+    """Represents an application team for a bot provided by Discord.
+
+
+    Attributes
+    -------------
+    id: :class:`int`
+        The team ID.
+    name: :class:`str`
+        The team name
+    icon: Optional[:class:`str`]
+        The icon hash, if it exists.
+    owner: :class:`User`
+        The team's owner ID
+    members: List[:class:`TeamMember`]
+        A list of the members in the team
+    """
+    __slots__ = ('_state', 'id', 'name', 'icon', 'owner_id', 'members')
+
+    def __init__(self, state, data):
+        self._state = state
+
+        self.id = int(data['id'])
+        self.name = data['name']
+        self.icon = data['icon']
+        self.owner = self._state.get_user(int(data['owner_user_id']))
```

Comment:
`owner` is not in the `__slots__` so this will fail.

```suggestion
        self.owner_id = int(data['owner_user_id'])
```

To bring back the `Team.owner` functionality you'd have to do something like this:

```py
@property
def owner(self):
    """Optional[:class:`User`]: The team's owner, if available from the cache."""
    return self._state.get_user(self.owner_id)
```